### DISCUSSION TITLE
--cluster-init in generated /etc/systemd/system/k3s.service

### DISCUSSION THREAD

**Discussion by udf2457** [k3s-io/k3s#7107] [Q&A] [answered]
I noticed that `--cluster-init` is present under `ExecStart` in the generated `/etc/systemd/system/k3s.service` file.

Is this intentional or an omission ?

**Comment by brandond** [comment]
I'm not sure I understand the question, what sort of omission would that be? The systemd unit is generated by the install script, so whatever args you passed to the installer will be in there.

**Comment by udf2457** [comment]
Well, yeah, "whatever args I passed", but what I'm saying is should I really  be passing `--cluster-init` again on start/restart via systemd ?  Seems a bit weird, no ?  Since a re-init is not desired ? i.e. should your install script not strip it out ?

**Reply by brandond** [reply]
When would it strip it out? The install script doesn't know whether or not you've already started K3s or not; it doesn't make any attempt to "correct" your args based on the state of the cluster.

Either way, it's harmless to leave the flag there on subsequent starts of the service. If the etcd datastore files exist on disk, the cluster is not re-initialized.

**Reply by zzhirong** [reply]
While usually harmless, leaving  `--cluster-init`  is harmful if the hostname changes and the host restarts. It causes the node to attempt joining as a new identity with the old data, which I just encountered.

**Reply by brandond** [reply, marked as answer]
This has nothing to do with --cluster-init. Kubernetes will **always** register a new node if the hostname changes, unless you've hardcoded the node-name so that it is not derived from the hostname.

**Reply by zzhirong** [reply]
Got it, thanks for the clarification. You're absolutely right, it's entirely due to the hostname change triggering a new registration.
